app/scripts/modules/core/src/task/TaskProgressBar.tsx

```diff
@@ -0,0 +1,79 @@
+import * as React from 'react';
+import { OverlayTrigger, Tooltip } from 'react-bootstrap';
+
+import { ITask } from 'core/domain';
+import { robotToHuman } from 'core/presentation/robotToHumanFilter/robotToHuman.filter';
+
+export interface ITaskProgressBarProps {
+  task: ITask;
+}
+
+export function TaskProgressBar(props: ITaskProgressBarProps) {
+  const {
+    task: { steps, id },
+    task,
+  } = props;
```

Comment:
Personal opinion: I find this destructure somewhat odd to read.  

How do you feel about this instead?
```js
const { task } = props;
const { steps, id } = task;
```

Reply:
please elaborate on 👎 @erikmunson

Reply:
😄 no strong opinion from my end (hence the reaction vs. commenting). I personally prefer the single assignment form in terms of readability, but it's deeply subjective. Also worth noting that there even _another_ option which is destructuring some or all of this in the function parameter itself rather than as a `const` assignment.

Reply:
I started with that, then tried making it a single line:
```js
const { task: { steps, id }, task } = props;
```
which I liked, but after `prettier` reformatted it, it does seem more confusing. Happy to go back to
```js
const { task } = props;
const { steps, id } = task;
```